src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1224,7 +1228,12 @@ private int GetSlotForMemberPostCondition(Symbol member)
 
         private void EnforceNotNullWhenForPendingReturn(PendingBranch pendingReturn, BoundReturnStatement returnStatement)
         {
-            var parameters = this.MethodParameters;
+            if (_symbol is not MethodSymbol method)
+            {
+                return;
+            }
+
+            ImmutableArray<ParameterSymbol> parameters = method.IsStatic ? method.Parameters : method.GetParametersIncludingExtensionParameter();
```

Comment:
> `method.IsStatic ? method.Parameters : method.GetParametersIncludingExtensionParameter()`

It feels like there could be a helper for this common pattern. Even better, the existing helper could have a non-optional parameter specifying what should happen in the static case so callers don't forget to handle that properly.